docs/source/basic.rst

```diff
@@ -116,6 +116,63 @@ notify their running by creating a .pid file.
 
 
 
+Timeouts
```

Comment:
> Maybe we should enhance timeout definition for docblock

What kind of enhance do you mean? I can do it, but what exactly?

> Also, I don't think there's a need to list how to use timeout in all executors, since it's basically the same usage, and same behaviour. 

The same usage, but not the same behavior. I can remove examples of code and put the text into the previously mentioned section. What do you think?

> Plus maybe we should separate Timeout header and Context manager to separate doc (advanced, or simply usage?), and keep basic as simple list of executors.

Good idea!